### PR TITLE
Add documentation for deprecated CMAC_CTX functions

### LINKED ISSUE #5539: Missing man page for deprecated public CMAC_* functions
After a fresh clone:
```
[murray@hgm openssl]$ grep CMAC doc/ -r
doc/man3/EVP_PKEY_CTX_ctrl.pod:and B<p2> is MAC key. This is used by Poly1305, SipHash, HMAC and CMAC.
[murray@hgm openssl]$ 
```

### PR DESCRIPTION
**Pull request by erbsland-dev**
Fixes #5539: Create a new manual page `CMAC_CTX.pod` documenting the deprecated `CMAC_CTX` functions and add the necessary build dependencies. This page includes function descriptions, usage details, and replacement suggestions with the `EVP_MAC` interface.

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/CMAC_CTX.pod [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+with the specified key, key length, and cipher type.
+Optionally, an B<ENGINE> can be provided.
+
+CMAC_Update() processes data to be included in the CMAC calculation.
+This function can be called multiple times to update the context with
+additional data.
+
+CMAC_Final() finalizes the CMAC calculation and retrieves the resulting
+MAC value. The output is stored in the provided buffer, and the length is
+stored in the provided variable.
```
Comment (mattcaswell) [inline doc/man3/CMAC_CTX.pod:80]: Important to note is that you can call `CMAC_Final` with a NULL output buffer, and the output length will still be stored, i.e. so you can call it twice - once to decide how big a buffer you need to allocate, then you allocate it and call it again.

**Review by mattcaswell — CHANGES_REQUESTED** [review]
Good work!

Please also update util/missingcrypto.txt to remove from that file the list of functions that you have now documented. I'm actually slightly surprised that the CI did not complain about not doing this.

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]
Good work!

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**